prov/verbs/src/verbs_rma.c

```diff
@@ -444,7 +444,9 @@ fi_ibv_xrc_rma_write_fast(struct fid_ep *ep_fid, const void *buf,
 
 	ep->base_ep.wrs->rma_wr.wr.rdma.remote_addr = addr;
 	ep->base_ep.wrs->rma_wr.wr.rdma.rkey = (uint32_t) key;
+#if VERBS_HAVE_XRC
 	ep->base_ep.wrs->rma_wr.qp_type.xrc.remote_srqn = ep->peer_srqn;
+#endif /* VERBS_HAVE_XRC */
```

Comment:
Maybe FI_IBV_SET_REMOTE_SRQN should be more generic to handle this case as well.

Reply:
Modified the macro to take a variable as an argument. I've also moved the assignment outside of the creation/assignment of the work request so that the macro can be generalized.